### PR TITLE
Enhancements

### LINKED ISSUE #105: TypeError: path must be a string
```
TypeError: path must be a string
    at TypeError (native)
    at Object.fs.access (fs.js:192:11)
    at /home/steel/.atom/packages/flow-ide/node_modules/atom-linter/lib/helpers.js:244:10
    at findCachedAsync (/home/steel/.atom/packages/flow-ide/node_modules/atom-linter/lib/helpers.js:243:18)
    at Object.<anonymous> (/home/steel/.atom/packages/flow-ide/lib/index.js:54:36)
    at [object Generator].next (native)
    at step (/home/steel/.atom/packages/flow-ide/lib/index.js:3:273)
    at /home/steel/.atom/packages/flow-ide/lib/index.js:3:443
    at Object.lint (/home/steel/.atom/packages/flow-ide/lib/index.js:3:99)
    at linters.forEach.promises.push.Promise.then._this2.emitter.emit.linter (/home/steel/.atom/packages/linter/lib/linter-registry.js:67:26)
    at /home/steel/.atom/packages/linter/lib/linter-registry.js:66:23
    at Set.forEach (native)
    at LinterRegistry.lint (/home/steel/.atom/packages/linter/lib/linter-registry.js:63:18)
    at /home/steel/.atom/packages/linter/lib/linter.coffee:110:16
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /home/steel/.atom/packages/linter/lib/editor-linter.js:20:20
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (/usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1067:27)
    at TextBuffer.module.exports.TextBuffer.save (/usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1037:19)
    at TextEditor.module.exports.TextEditor.save (/usr/share/atom/resources/app.asar/src/text-editor.js:602:26)
    at Pane.module.exports.Pane.saveItem (/usr/share/atom/resources/app.asar/src/pane.js:534:18)
    at Pane.module.exports.Pane.saveActiveItem (/usr/share/atom/resources/app.asar/src/pane.js:517:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (/usr/share/atom/resources/app.asar/src/workspace.js:648:35)
    at atom-workspace.commandRegistry.add.core:save (/usr/share/atom/resources/app.asar/src/register-default-commands.js:173:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

Seems like we're caching even the null result of atom-linter::findCachedAsync, we shouldn't cache it when it's null

### LINKED ISSUE #107: Verify validateEditor on Atom <1.4.0
See https://github.com/AtomLinter/linter-eslint/issues/429 for details, we need to verify the changes in https://github.com/AtomLinter/atom-linter/pull/102 still work for Atom <1.4.0.

### PR DESCRIPTION
**Pull request by steelbrain**
Purposes of this PR

 - Fix a bug in findCachedAsync where it would cache and afterwards fail if file is not found
 - Try to use `async`/`await` to make codebase cleaner
 - Add checks in multiple places to avoid future possible bugs
 - Fixes an invalid check in text editor validation

Fixes #107
Fixes #105

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .idea
 node_modules
+/lib
```
Comment (Arcanemagus) [inline .gitignore]: Just `lib` will work.

File: spec/helper-spec.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import * as fs from 'fs'
 import * as path from 'path'
-const helpers = require('../lib/helpers')
+const helpers = require('../src/helpers')
```
Comment (Arcanemagus) [inline spec/helper-spec.js]: Shouldn't the specs still be testing the compiled code?

Comment (Arcanemagus) [inline spec/helper-spec.js]: Just checked the relevant commit, CI should be building before running the specs.

File: .gitignore (hunk 1 above)
Comment (steelbrain) [inline .gitignore]: Okie tokie

**Comment by steelbrain** [comment]
@Arcanemagus I tried to make transpiling on CI work but it seems like they have super-old node so I'm not gonna bother. Besides why do all this when atom's spec runner has builtin babel transpilation?

**Comment by steelbrain** [comment]
Stupid AppVeyor, anyways, this is ready for a review

cc @Arcanemagus

**Comment by Arcanemagus** [comment]
If Atom's specs transpile the babel code, why do we need to transpile it in the first place?

**Comment by steelbrain** [comment]
@Arcanemagus Because this isn't an atom module, this is an atom module dependency. They transpile the code for execution thinking it's an atom package, but we're gonna publish it to apm not npm. 

That's why we're transpiling the code.

**Comment by steelbrain** [comment]
Ping @Arcanemagus

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,274 @@
+'use babel'
+
+/* @flow */
+
+import * as Path from 'path'
+import * as FS from 'fs'
+import {
```
Comment (Arcanemagus) [inline src/index.js]: Not really against explicitly listing all of these, but any reason you didn't just do `import * from './helpers'`?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+  return [
+    [lineNumber, colStart],
+    [lineNumber, lineLength]
+  ]
+}
+
+export async function findAsync(directory: string, name: string | Array<string>): Promise<?string> {
+  validateFind(directory, name)
+  const names = [].concat(name)
```
Comment (Arcanemagus) [inline src/index.js]: I like how the previous code showed what this was accomplishing, but this is probably faster so :+1:.

**Comment by Arcanemagus** [comment]
LGTM, pending the `waitsForPromiseRejection` in specs.

**Pull request merged**